plugins/catalog-react/src/hooks/useEntityEnvironment.tsx

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useApi } from '@backstage/core-plugin-api';
+import React, {
+  PropsWithChildren,
+  createContext,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+import { catalogApiRef } from '../api';
+import useAsync from 'react-use/lib/useAsync';
+import {
+  ComponentEntityV1alpha1,
+  EnvironmentEntityV1alpha1,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import { useEntity } from './useEntity';
+
+type EntityEnvironmentContextProps = {
+  environments?: EnvironmentEntityV1alpha1[];
+  environment?: EnvironmentEntityV1alpha1;
+  setEnvironment: (environment: EnvironmentEntityV1alpha1 | undefined) => void;
+  loading: boolean;
+  error?: Error;
+};
+
+/**
+ * Creates new context for entity listing and filtering.
+ * @public
+ */
+export const EntityEnvironmentContext = createContext<
+  EntityEnvironmentContextProps | undefined
+>(undefined);
+
+/**
+ * Provides entities and filters for a catalog listing.
+ * @public
+ */
+export const EntityEnvironmentProvider = (props: PropsWithChildren<{}>) => {
```

Comment:
Basic idea is that you can switch between environments on the front end to see information for a specific environment. This still needs more tailoring for an entity-by-entity experience, like only showing options relevant the current entity.